var/spack/repos/builtin/packages/py-numba/package.py

```diff
@@ -23,27 +24,28 @@ class PyNumba(PythonPackage):
     version("0.48.0", sha256="9d21bc77e67006b5723052840c88cc59248e079a907cc68f1a1a264e1eaba017")
     version("0.40.1", sha256="52d046c13bcf0de79dbfb936874b7228f141b9b8e3447cc35855e9ad3e12aa33")
 
-    depends_on("python@3.7:3.10", when="@0.55.0:", type=("build", "run"))
+    depends_on("python@3.8:3.12", when="@0.57", type=("build", "run"))
+    depends_on("python@3.7:3.11", when="@0.56", type=("build", "run"))
+    depends_on("python@3.7:3.10", when="@0.55", type=("build", "run"))
     depends_on("python@3.7:3.9", when="@0.54", type=("build", "run"))
     depends_on("python@3.6:3.9", when="@0.53", type=("build", "run"))
     depends_on("python@3.6:3.8", when="@0.52", type=("build", "run"))
-    # set upper bound for python the same as newer release
     depends_on("python@3.6:3.8", when="@0.48:0.51", type=("build", "run"))
     depends_on("python@3.3:3.7", when="@0.40.1:0.47", type=("build", "run"))
-    depends_on("py-numpy@1.18:1.23", when="@0.56.1:", type=("build", "run"))
+    depends_on("py-numpy@1.21:1.25", when="@0.57.0:", type=("build", "run"))
+    depends_on("py-numpy@1.18:1.23", when="@0.56.1:0.56.4", type=("build", "run"))
     depends_on("py-numpy@1.18:1.22", when="@0.55.2:0.56.0", type=("build", "run"))
     depends_on("py-numpy@1.18:1.21", when="@0.55.0:0.55.1", type=("build", "run"))
     depends_on("py-numpy@1.17:1.20", when="@0.54", type=("build", "run"))
-    # set upper bound for py-numpy the same as newer release
     depends_on("py-numpy@1.15:1.20", when="@0.48:0.53", type=("build", "run"))
     depends_on("py-numpy@1.10:1.20", when="@:0.47", type=("build", "run"))
     depends_on("py-setuptools", type=("build", "run"))
-
-    depends_on("py-llvmlite@0.39", when="@0.56", type=("build", "run"))
-    depends_on("py-llvmlite@0.38", when="@0.55", type=("build", "run"))
-    depends_on("py-llvmlite@0.37", when="@0.54.0", type=("build", "run"))
-    depends_on("py-llvmlite@0.34", when="@0.51.1", type=("build", "run"))
-    depends_on("py-llvmlite@0.33", when="@0.50.1", type=("build", "run"))
+    depends_on("py-llvmlite@0.40:0.41", when="@0.57", type=("build", "run"))
+    depends_on("py-llvmlite@0.39:0.40", when="@0.56", type=("build", "run"))
+    depends_on("py-llvmlite@0.38:0.39", when="@0.55", type=("build", "run"))
+    depends_on("py-llvmlite@0.37:0.38", when="@0.54.0", type=("build", "run"))
+    depends_on("py-llvmlite@0.34:0.35", when="@0.51.1", type=("build", "run"))
+    depends_on("py-llvmlite@0.33:0.34", when="@0.50.1", type=("build", "run"))
```

Comment:
The upper bounds were already there because `py-llvmlite` was restricted to one minor version.
In addition the upper bound are exclusive in setup.py, so these new ranges are not correct (see [setup.py#L379](https://github.com/numba/numba/blob/0.57.0/setup.py#L370))

Reply:
Sorted, thanks. I was going by the version spec at the top of the `setup.py` - hadn't realised they were then not using them via `>=min <max`.